template/task-templates.wdl

```diff
@@ -116,6 +116,53 @@ task use_all_cores_task {
     }
 }
 
+task localize_files_task {
+    meta {
+        description: "This template is appropriate for tasks which assume multiple files share the same basename with specific extensions and/or that these files are in the same directory (this task will use BAM and BAI files as an example)." 
+    }
+
+    parameter_meta {
+        bam: "Input BAM format file to <brief description of task>"
+        bam_index: "BAM index file corresponding to the input BAM"
+        memory_gb: "RAM to allocate for task, specified in GB"
+        disk_size_gb: "Disk space to allocate for task, specified in GB"
+        max_retries: "Number of times to retry in case of failure"
+    }
+
+    input {
+        File bam
+        File bam_index
+        Int memory_gb = <>
+        Int disk_size_gb = <>
+        Int max_retries = 1
+    }
+
+    command <<<
+        set -euo pipefail
+
+        # localize BAM and BAI to CWD
+        # some backends place them in separate directories
+        # some backends prevent writing to the inputs directories
+        # to accomodate these possibilites, create symlinks in CWD
```

Comment:
Hmmm.. Maybe this comment contains incorrect information, but we do need the code change. `miniwdl` doesn't let you write to the input directory (which I discovered by passing the wrong BAI to tasks) so our old code wouldn't work.

I have a not so detailed memory of reading a GitHub issue for a WDL runnner (of course I can't remember which) where the user was complaining about their BAM and BAI not being in the same directory.

I'm open to changing this comment to maybe be less specific (or more true), but the code change is needed.

Reply:
I see. I went back to see where you were using it. The backend makes the input directory read-only. So since we're trying to create/rename the bam index, it won't work there. Hence the need to localize. I think we can just remove the comment about separate directories and it should be correct.